tools-local/doc.tasks/GenerateMarkdownDoc.cs

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Xml.XPath;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace NuGetTasks
+{
+    public class GenerateMarkdownDoc : Task
+    {
+        /// <summary>
+        /// Project files to process
+        /// </summary>
+        public ITaskItem[] ProjectFiles { get; set; }
+        /// <summary>
+        /// Filepath to the generated documentation
+        /// </summary>
+        public ITaskItem OutputFile { get; set; }
+        /// <summary>
+        /// Repository Url to create links to the file
+        /// </summary>
+        public string GitHubRepositoryUrl { get; set; }
+        /// <summary>
+        /// Repository path on disk to relativize paths
+        /// </summary>
+        public string RepositoryRoot { get; set; }
+        /// <summary>
+        /// Title for the document
+        /// </summary>
+        public string Title { get; set; } = "NuGet Project Overview";
+
+        public override bool Execute()
+        {
+            var outputFilePath = OutputFile.GetMetadata("FullPath");
+
+            var dirname = Path.GetDirectoryName(outputFilePath);
+            Directory.CreateDirectory(dirname);
+            Log.LogMessage(MessageImportance.Low, $"Created directories: {dirname}");
+
+            Log.LogMessage(MessageImportance.Low, $"Creating documentation: {outputFilePath}");
```

Comment:
Maybe it's a good idea to write a basically intro to the overview and then have 2 categories of projects in the list. 
Specifically have NuGet.Clients and NuGet.Core.
But that can be done later. 
We can make incremental changes.

Reply:
Added tests projects to the overview.

Reply:
Added test projects to the list.

Reply:
Also, I added a brief description in project-list.md